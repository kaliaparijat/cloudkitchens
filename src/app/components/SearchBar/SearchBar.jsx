import React from 'react';
import { withStyles } from '@material-ui/core/styles';
import Radio from '@material-ui/core/Radio';

const SearchBar = (props) => {
  return (
    <form className="searchBar">
      <label htmlFor="cooked">Get all cooked orders in past <input
        type="text"
        name="cooked"
        defaultValue={props.searchText}
      /> seconds</label>
      <p>
        <Radio
          name="filter"
          value="active"
          onChange={props.searchByCriteria}
          checked={!props.isCooking && !props.isHistorical}
          defaultChecked
        />
        <label htmlFor="active"
               name="activeOrders-label"
        >
          Active orders
        </label>
        <Radio
          name="filter"
          value="cooking"
          checked={props.isCooking}
          onChange={props.searchByCriteria}
        />
        <label htmlFor="cooking"
               name="cooking-label"
        >
          Cooking now
        </label>
        <Radio
          name="filter"
          value="historical"
          checked={props.isHistorical}
          onChange={props.searchByCriteria}
        />
        <label htmlFor="historical"
               name="historical-label"
        >
          All past orders
        </label>
      </p>
    </form>
  )
};

export default SearchBar;