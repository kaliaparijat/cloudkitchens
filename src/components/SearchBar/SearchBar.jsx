import React from 'react';

class SearchBar extends React.Component {
    constructor(props){
        super(props);
        this.filterByCookingNow = false;
        this.filterByHistoricalOrders = false;
        this.searchText = '';
        this.toggleFilterByCookingNow = this.toggleFilterByCookingNow.bind(this);
    }

    toggleFilterByCookingNow = () => {
        this.filterByCookingNow = true;
        if(this.filterByCookingNow) {
            this.filterByHistoricalOrders = false;
            this.searchText = '';
        }
    }

    render() {
        return (
            <form className="searchBar">

                <label htmlFor="cooked">Get all cooked orders in past <input
                    type="text"
                    name="cooked"
                    readOnly
                    value={this.searchText}
                /> seconds</label>
                <p>
                    <input
                        type="checkbox"
                        name="cooking"
                        onChange={this.toggleFilterByCookingNow}

                    />
                    <label htmlFor="cooking"
                           name="cooking-label"
                    >
                        Orders currently cooking
                    </label>
                    <input
                        type="checkbox"
                        name="historical"
                        readOnly
                        checked={this.filterByHistoricalOrders}
                    />
                    <label htmlFor="historical"
                           name="historical-label"
                    >
                        All processed orders
                    </label>
                </p>
            </form>
        )
    }
}

export default SearchBar;