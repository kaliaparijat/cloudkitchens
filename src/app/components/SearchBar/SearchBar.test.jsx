import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import SearchBar from './SearchBar';

describe('SearchBar',() => {
  let searchBar;

  beforeAll(() => {
    searchBar = render(<SearchBar searchText="33" isHistorical={false} isCooking={false} searchByCriteria={jest.fn()} />)
  });

  it('should render with prop values', () => {
    const { asFragment } = searchBar;
    expect(asFragment()).toMatchSnapshot();
  });

});

