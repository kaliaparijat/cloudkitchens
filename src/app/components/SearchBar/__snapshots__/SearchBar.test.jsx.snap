// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar should render with prop values 1`] = `
<DocumentFragment>
  <form
    class="searchBar"
  >
    <label
      for="cooked"
    >
      Get all cooked orders in past 
      <input
        name="cooked"
        type="text"
        value="33"
      />
       seconds
    </label>
    <p>
      <input
        name="filter"
        type="radio"
        value="cooking"
      />
      <label
        for="cooking"
        name="cooking-label"
      >
        Cooking now!
      </label>
      <input
        name="filter"
        type="radio"
        value="historical"
      />
      <label
        for="historical"
        name="historical-label"
      >
        All orders
      </label>
    </p>
  </form>
</DocumentFragment>
`;
